TLM/TLM/UI/SubTools/LaneConnectorTool.cs

```diff
@@ -1296,12 +1329,87 @@ private class LaneEnd {
                 cutEnd: 0,
-                minY: -1f,
-                maxY: 1280f,
-                renderLimits: false,
+                minY: middlePoint.y - overdrawHeight,
+                maxY: middlePoint.y + overdrawHeight,
+                renderLimits: renderLimits,
                 alphaBlend: true);
         }
 
+        /// <summary>
+        /// Draw accurate bezier line with option to be visible through terrain and other objects or not.
+        /// Lane rendering mesh(box) has very low height which prevents overdraw and other performance issues
+        /// </summary>
+        /// <param name="cameraInfo">Camera instance to use</param>
+        /// <param name="bezier">Bezier acr to render</param>
+        /// <param name="color">Color</param>
+        /// <param name="outlineColor">Outline color</param>
+        /// <param name="size">Bezier line thickness</param>
+        /// <param name="underground">Should be visible through obstacles like terrain or other objects</param>
+        private void DrawLaneCurve(RenderManager.CameraInfo cameraInfo,
+                                   ref Bezier3 bezier,
+                                   Color color,
+                                   Color outlineColor,
+                                   float size = 0.08f,
+                                   bool underground = false) {
+            Bounds bounds = bezier.GetBounds();
+            // Draw black outline
+            RenderManager.instance.OverlayEffect.DrawBezier(
+                cameraInfo: cameraInfo,
+                color: outlineColor,
+                bezier: bezier,
+                size: size * 1.5f,
+                cutStart: 0,
+                cutEnd: 0,
+                minY: bounds.min.y - 0.5f,
+                maxY: bounds.max.y + 0.5f,
+                renderLimits: underground,
+                alphaBlend: false);
+
+            // Inside the outline draw colored bezier
+            RenderManager.instance.OverlayEffect.DrawBezier(
+                cameraInfo: cameraInfo,
+                color: color,
+                bezier: bezier,
+                size: size,
+                cutStart: 0,
+                cutEnd: 0,
+                minY: bounds.min.y - 0.5f,
+                maxY: bounds.max.y + 0.5f,
+                renderLimits: underground,
+                alphaBlend: true);
+        }
+
+        /// <summary>
+        /// Calculates accurate bezier arc between two lane ends
+        /// </summary>
+        /// <param name="sourceLaneEnd">Start position marker</param>
+        /// <param name="targetLaneEnd">End position marker</param>
+        /// <returns>Bezier arc</returns>
+        private Bezier3 CalculateBezierConnection(LaneEnd sourceLaneEnd, LaneEnd targetLaneEnd) {
+            Bezier3 bezier3 = default;
+            bezier3.a = sourceLaneEnd.NodeMarker.Position;
+            bezier3.d = targetLaneEnd.NodeMarker.Position;
+
+            Vector3 sourceLaneDirection =
```

Comment:
Are you sure you don't need to normalize the result of the tangent?
I remember it is not a unit vector

Reply:
Look at 1403 and 1405, actually I can move that property call since the variable name is slightly wrong now (like you said it's not unit vector yet)